### ISSUE TITLE
Incorrect handling of HTTP methos other then GET and POST.

### ISSUE THREAD

**Issue by pacesm** [EOxServer/eoxserver#359]
The EOxServer OWS services respond incorrectly to methods other then GET or POST. 

1) The proper response to request with unsupported method should be `405 Method Not Allowed` [[see e.g.](https://www.w3.org/Protocols/rfc2616/rfc2616-sec10.html)] but the server responds with `400 Bad Request` and the following OWS exception:

```
<?xml version='1.0' encoding='iso-8859-1'?>
<ows:ExceptionReport xmlns:ows="http://www.opengis.net/ows/2.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
  <ows:Exception exceptionCode="NoApplicableCode">
    <ows:ExceptionText>'NoneType' object has no attribute 'version'</ows:ExceptionText>
  </ows:Exception>
</ows:ExceptionReport>
```

(which seems to me rather like a `500 Internal Server Error`, see related issue #360)

2) The server should handle correctly the CORS pre-flight checks (OPTIONS method). As you can read, e.g., [here in Sec. Preflighted requests](https://developer.mozilla.org/en-US/docs/Web/HTTP/Access_control_CORS), the CORS-compliant browsers will always send an OPTION request before sending XML payload via POST.

The EOxServer ignores the request content type of the POST payload and the OPTIONS requests can be easily avoided by declaring the XML as plain text. Despite the fast that this issue is not critical the behaviour of the server is not correct.

**Comment by constantinius** [comment]
Yes, 1) should be a `500` with that error message.
For 2): we could respond with a `200` for all `OPTIONS` request, if I interpret this right. I don't know any other use of that method. Would this be feasible?

**Comment by pacesm** [comment]
add 1)
you are right that these are two issues. A crash like this should lead to a `500` response. But the unsupported method should lead a `405` error. 

add 2)
IMO, the response to OPTIONS should be an empty `200` response with the relevant HTTP headers set (such as CORS allowed methods and allowed headers; handled either by the Apache or Django middleware). The OPTION handler can ignore a possible query string or request body as the headers are the same for the whole OWS end-point.

**Comment by constantinius** [comment]
Would you say that 8db797a fixes the issue?

**Comment by pacesm** [comment]
When I look at the code, the `OPTIONS` requests should be fine. It is not obvious though what happens when the server receives, .e.g., `PUT` request (should be plain a `405`). 

However, when I quickly deployed the branch and I'm still getting the `400` and `'NoneType' object has no attribute 'version'` OWS exception for anything else than `GET` or `POST`.

**Comment by pacesm** [comment]
The issue is solved by merging PR #364.

The related internal server error reported as wrong requests is covered by another issue (see #360).
